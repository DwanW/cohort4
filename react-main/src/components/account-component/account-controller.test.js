import React from 'react';
import { shallow } from 'enzyme';
import AccountController from './account-controller';

describe('AccountController Component test', () => {
    let wrapper;
    let initialState = {
        accArr: [],
        newAccName: '',
        newAccBalance: 0,
        hideAccPrompt: true,
        hideInfoPrompt: true,
        total: 0,
        maxAcc: '',
        minAcc: '',
        info:'Account Not Found'
    }

    beforeEach(()=> {
        wrapper = shallow(<AccountController />)
    })
    
    it('should render AccountController Component', ()=> {
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.state()).toEqual(initialState);
    });

    it('should render the correct info-panel', ()=> {
        expect(wrapper.find('#info-panel').text()).toBe("Account Not Found");
    });

    it('should correctly hide or display info-panel', ()=>{
        expect(wrapper.find('.name-prompt').at(0).hasClass('hide')).toBe(true);
        expect(wrapper.find('.name-prompt').at(1).hasClass('hide')).toBe(true);
        wrapper.find("#addAcc").simulate('click');
        expect(wrapper.find('.name-prompt').at(0).hasClass('hide')).toBe(false);
        wrapper.find("#createAcc").simulate('click');
        expect(wrapper.find('.name-prompt').at(0).hasClass('hide')).toBe(true);
        wrapper.find("#getTotal").simulate('click');
        expect(wrapper.find('.name-prompt').at(1).hasClass('hide')).toBe(false);
    });
});