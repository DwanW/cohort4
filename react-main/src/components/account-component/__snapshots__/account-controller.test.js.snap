// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountController Component test should render AccountController Component 1`] = `ShallowWrapper {}`;
